proofs/src/lib.rs

```diff
@@ -0,0 +1,316 @@
+        webpki::Time::from_seconds_since_unix_epoch(validation_time),
+    )
+    .map_err(AttestError::InvalidCertChain)?;
+
+    Ok(())
+}
+
+// Check that cose sign1 structure is signed with the key in the end
+// entity certificate.
+fn verify_cose_sign1_sig(
+    end_entity_certificate: &[u8],
+    cose_sign1: &CoseSign1,
+) -> Result<(), AttestError> {
+    use x509_cert::der::Decode;
+
+    let ee_cert = x509_cert::certificate::Certificate::from_der(end_entity_certificate)
+        .map_err(|_| AttestError::FailedToParseCert)?;
+
+    // Expect v3
+    if ee_cert.tbs_certificate.version != x509_cert::certificate::Version::V3 {
+        return Err(AttestError::InvalidEndEntityCert);
+    }
+
+    let pub_key = ee_cert
+        .tbs_certificate
+        .subject_public_key_info
+        .subject_public_key;
+    let key =
+        PublicKey::from_sec1_bytes(pub_key).map_err(|_| AttestError::FailedDecodeKeyFromCert)?;
+    let key_wrapped = P384PubKey(key);
+
+    // Verify the signature against the extracted public key
+    let is_valid_sig = cose_sign1
+        .verify_signature::<Sha2>(&key_wrapped)
+        .map_err(|_| AttestError::InvalidCOSESign1Signature)?;
+    if is_valid_sig {
+        Ok(())
+    } else {
+        Err(AttestError::InvalidCOSESign1Signature)
+    }
+}
+
+struct P384PubKey(p384::PublicKey);
+impl SigningPublicKey for P384PubKey {
+    fn get_parameters(&self) -> Result<(SignatureAlgorithm, MessageDigest), CoseError> {
+        Ok((SignatureAlgorithm::ES384, MessageDigest::Sha384))
+    }
+
+    fn verify(&self, digest: &[u8], signature: &[u8]) -> Result<bool, CoseError> {
+        let signature_wrapped =
+            Signature::try_from(signature).map_err(|e| CoseError::SignatureError(Box::new(e)))?;
+
+        let verifier = VerifyingKey::from(self.0);
+        verifier
+            .verify_prehash(digest, &signature_wrapped)
+            .map(|_| true)
+            .map_err(|e| CoseError::SignatureError(Box::new(e)))
+    }
+}
+
+struct Sha2;
+impl Hash for Sha2 {
+    fn hash(digest: MessageDigest, data: &[u8]) -> Result<Vec<u8>, CoseError> {
+        use sha2::Digest as _;
+        match digest {
+            MessageDigest::Sha256 => Ok(sha2::Sha256::digest(data).to_vec()),
+            MessageDigest::Sha384 => Ok(sha2::Sha384::digest(data).to_vec()),
+            MessageDigest::Sha512 => Ok(sha2::Sha512::digest(data).to_vec()),
+        }
+    }
+}
+
+pub fn parse_and_verify_aws_nitro_attestation<S: AsRef<str>>(
+    encoded_attestation: S,
+) -> Result<AttestationDoc, AttestError> {
+    // Decode the base64 string
+    let decoded_bytes = base64::decode(encoded_attestation.as_ref())
+        .map_err(|e| AttestError::Base64DecodingError(e.to_string()))?;
+
+    let trusted_root_certificate = cert_from_pem(AWS_ROOT_CERT_PEM).unwrap();
+
+    let current_time = std::time::SystemTime::now()
+        .duration_since(std::time::UNIX_EPOCH)
+        .unwrap()
+        .as_secs();
+
+    attestation_doc_from_der(
```

Comment:
should we add a comment indicating this function is also responsible for the `verify` step as well? we can also rename it if it adds clarity. either way, quite nitty!

Reply:
Addressed in https://github.com/tkhq/rust-sdk/pull/18/commits/5ecfbef882ef89ec5c77b2c2c18b7f820dfaf59c